docs/blog/posts/reworking-names.md

```diff
@@ -0,0 +1,49 @@
+---
+authors:
+  - lucas
+title: "Enhanced Representation and Management of Global Identifiers in hax"
+date: 2025-03-19
+---
+
+# Enhanced Representation and Management of Global Identifiers in hax
+
+In this technical overview, we present significant enhancements to the internal architecture used by the hax engine for representing and manipulating global identifiers. These updates address fundamental design limitations, greatly improving maintainability, clarity, and backend code generation precision.
+
+## Motivation for the Improvements
```

Comment:
Rename to "Global Identifiers in Rust" explaining where all the global identifiers come from

At the end of this section, you can have the comment about a new category of global identifiers, which is introduced by hax as part of the engine.